### DISCUSSION TITLE
Anchored via/shaping trackpoints

### DISCUSSION THREAD

**Discussion by 0709wiwiwi** [devemux86/cruiser#477] [Q&A] [answered]
Anchor shaping trackpoints.

The Intelligent routing mode allows you to only make changes on individual track sections of imported (kurvig) tracks.
Depending on how far apart the shaping or via points are located, it can undesirably change rather long track sections.
So Leeb's question was to quickly place a shaping point which you can then freely move to a new position for a change.

Place this shaping point automatically on the nearest existing trkpt that so is promoted into a shaping trackpoint. 
If you don't move this newly created shaping trackpoint after all, no partial recalculation is triggered.
Than promote a second existing trackpoint into a shaping trackpoint in the same way.
This way, a new short track section is created. Modify this newly created shorter tracksection as desired.

**Comment by devemux86** [comment]
@0709wiwiwi thanks for the suggestion.

This is how it will work: if long press on the route line and select set via / shaping point,
then the crosshair appears snapped on the route line waiting to be moved or press OK.

>  If you don't move this newly created shaping trackpoint after all, no partial tracksection recalculation so should be triggered.

This involves both via and shaping points. I will have to check how it is possible. :slightly_smiling_face:

**Comment by 0709wiwiwi** [comment]
>  I will have to check how it is possible.
 
Sure Emux, caution is sure needed not to disrupt the already fine planner operations 
So I went over the quick procedure a bit more and still think it can be done safely.

Cruiser current.
Press near a track. Choose shaping or via point. Visually, the shaping (via) point is away from the road. 
However, an additional associated shaping track point is immediately created by a router recalculation.

NEW Leeb fast.  (shaping point by default)
**Press right onto the track.** Cruiser than snaps a cross hair **shaping** point onto the nearest track point. 
Confirm without moving, promotes this track point into a **shaping** track point without any recalculation. 
Or an even shorter Leeb procedure given you can change **shaping** points to via points any time later. 
If you do not move the crosshair shaping point by pressing ok no new calculation should so happen. 
Moving the crosshair shaping pont than confirm the new position by press ok a recalculation happens.
In a navigation track, by the way, you will usually find a lot more **shaping** points than via points.

**Reply by devemux86** [reply]
Changing waypoint types between via and shaping already does not recalculate the route in Cruiser.

It will be possible, but do not want to over complicate it.
If we set a via / shaping point near the route, it should be added instantly without any calculation.
If it is not near the route, a section calculation will happen.

**Comment by devemux86** [comment]
> If you don't move this newly created shaping trackpoint after all, no partial tracksection recalculation so should be triggered.

Adding via / shaping points _on the route_ will work without route calculation in the next Cruiser.

**Reply by devemux86** [reply]
Adding via / shaping points on the route without route calculation is available for testing in:
- https://github.com/devemux86/cruiser/discussions/482

**Reply by 0709wiwiwi** [reply]
The test was a succes. Add a small deviation onto a kurviger generated traject. The rest of the original design remains intact.
![Change an existing kurviger traject](https://github.com/devemux86/cruiser/assets/54167201/b7d45126-07c6-40a3-b5f2-5dcfd097ccc5)

**Comment by devemux86** [comment, marked as answer]
Adding via / shaping points on the route without route calculation is available in [Cruiser 4.0.0](https://play.google.com/store/apps/details?id=gr.talent.cruiser).
